driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/syncadapter/SingleResultSubscriber.java

```diff
@@ -17,36 +17,29 @@
 package com.mongodb.reactivestreams.client.syncadapter;
 
 import com.mongodb.MongoInterruptedException;
+import com.mongodb.MongoTimeoutException;
 import com.mongodb.lang.Nullable;
 import org.reactivestreams.Subscriber;
 import org.reactivestreams.Subscription;
 
-import java.util.ArrayList;
-import java.util.List;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 
-class SyncSubscriber<T> implements Subscriber<T> {
-    private final List<T> results = new ArrayList<>();
+class SingleResultSubscriber<T> implements Subscriber<T> {
+    private volatile T result;
     private volatile Throwable exception;
     private final CountDownLatch latch = new CountDownLatch(1);
 
     @Nullable
-    T first() {
-        List<T> all = all();
-        if (all.isEmpty()) {
-            return null;
-        }
-        return all.get(0);
-    }
-
-    List<T> all() {
+    T get() {
         try {
-            latch.await(10, TimeUnit.SECONDS);
+            if (!latch.await(30, TimeUnit.SECONDS)) {
```

Comment:
Increased timeout to workaround the async server selection bug I ran into with a 10 second timeout.